pkg/image/qemu.go

```diff
@@ -1,16 +1,118 @@
+/*
+Copyright 2018 The CDI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
 package image
 
 import (
-	"os/exec"
+	"encoding/json"
+	"fmt"
+	"net/url"
 
+	"github.com/golang/glog"
 	"github.com/pkg/errors"
+
+	"kubevirt.io/containerized-data-importer/pkg/system"
 )
 
-func ConvertQcow2ToRaw(src, dest string) error {
-	cmd := exec.Command("qemu-img", "convert", "-f", "qcow2", "-O", "raw", src, dest)
-	err := cmd.Run()
+const (
+	networkTimeoutSecs = 3600    //max is 10000
+	maxMemory          = 1 << 30 //value from OpenStack Nova
+	maxCPUSecs         = 30      //value from OpenStack Nova
+)
+
+// QEMUOperations defines the interface for executing qemu subprocesses
+type QEMUOperations interface {
+	ConvertQcow2ToRaw(string, string) error
+	ConvertQcow2ToRawStream(*url.URL, string) error
+	Validate(string, string) error
+}
+
+type qemuOperations struct{}
+
```

Comment:
Something to consider - interface implementation can be checked at compile time with

`var _ qemuOperations = QEMUOperations` 

to ensure that changes in the interface definition are reflected by `qemuOperations`